plugins/mcreator-core/blockly/js/mcreator_extensions.js

```diff
@@ -37,6 +37,30 @@ Blockly.Extensions.register('is_custom_loop',
         Blockly.libraryBlocks.loops.loopTypes.add(this.type);
     });
 
+Blockly.Extensions.register('procedure_dependencies_onchange_mixin',
+    function () {
+        this.setOnChange(function (changeEvent) {
+            // Trigger the change only if a block is changed or created
+            if (changeEvent.type !== Blockly.Events.BLOCK_CHANGE &&
+                changeEvent.type !== Blockly.Events.BLOCK_CREATE ||
+                changeEvent instanceof InputCheckChange) {
+                return;
+            }
+            const group = Blockly.Events.getGroup();
+            // Makes it so the block change and the unplug event get undone together.
+            Blockly.Events.setGroup(changeEvent.group);

```

Comment:
I have just checked and setting values of our custom fields via `callback` functions does not set the appropriate group for the event triggered after the value change, which does not allow to undo these `InputCheckChange` events together with updating field values. @AlsoSomeoneElse, could you please take a look at this?

Reply:
Nevermind, seems I got blind another time.